### PR TITLE
2017/2018 Content updates for UK Full time students

### PR DESCRIPTION
**Pull request by ikennaokpala**
[Trello card](https://trello.com/c/AhBxmn0G/479-8-student-form-finder-add-2017-18-remove-2015-16)

## Motivation

The need to update the student form finder to end application for 2015/2016 session and  add/inform them about 2017/2018 applications.

### *At the moment, some of the file sizes for the PDFs are not available, before merging this PR make sure to add file sizes.*

See production issue re-produced [here](https://smart-answers-pr-2877-bug.herokuapp.com/student-finance-forms/y/uk-full-time/apply-loans-grants) that is fixed by 0862b05. Don't forget to refresh to reproduce this issue.

## Factcheck

[Preview link](https://smart-answers-pr-2877.herokuapp.com/student-finance-forms/y/uk-full-time/apply-loans-grants)

## Expected changes

[GOV.UK](https://gov.uk/student-finance-forms/y/uk-full-time/apply-loans-grants)
- Remove (almost) all 2015 to 2016 outcomes for full-time students
- Add 2017 to 2018 outcomes for full-time students
    - Add 6 new outcomes
    - Amend 3 existing outcomes

2016 to 2017 full-time outcomes are unaffected.
Part-time and EU outcomes are unaffected.

### Before

screen shot 2017-01-11 at 11 36 26

### After

screen shot 2017-01-11 at 11 36 09

Also contains related changes from #2845

### CODE REVIEW

File: lib/smart_answer_flows/student-finance-forms.rb [hunk 1]
```diff
@@ -73,8 +73,12 @@ def define
...
 
       multiple_choice :what_year_full_time? do
-        option 'year-1617'
-        option 'year-1516'
+        question_value = self
+        precalculate :setup_options do
+          question_value.option "year-1718" if type_of_student == "uk-full-time"
+          question_value.option "year-1617"
+          question_value.option "year-1516" if type_of_student == "eu-full-time"
+        end
```
Comment (leenagupte) [inline lib/smart_answer_flows/student-finance-forms.rb]: @ikennaokpala The commit heading needs to be changed from `2105/16` to `2015/16' :)

I would have split this into two commits:

1) Add `2017-2018` as an option for `uk-full-time` students, and adds new outcome templates that match the option name and adds new test artefacts.
2) Remove `2015-2015` as an option for `uk-full-time` students and removes test artefacts.

That would make it easier to follow what's going on a bit.

**Comment by chao-xian** [comment]
From my point of view things looked fine @ikennaokpala but with caveat of my trying to get my head around it. I'll leave you to address @leenagupte's comments.

File: lib/smart_answer_flows/student-finance-forms.rb (hunk 1 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/student-finance-forms.rb]: I will try out your suggestion.. thanks

Comment (ikennaokpala) [inline lib/smart_answer_flows/student-finance-forms.rb]: I have split the commits ea0fb4ca66563f946e5558437765401422144ccb, f0026fc17fdd452438e601d89fbd5ccc66f5f660

**Comment by ikennaokpala** [comment]
@leenagupte You thoughts have been considered and articulated in this [commit note](https://github.com/alphagov/smart-answers/pull/2877/commits/ea0fb4ca66563f946e5558437765401422144ccb)

Splitting ```what_year_full_time?``` to ```what_year_full_time_uk?`` and ```what_year_full_time_eu?``` is a good idea, but involves the extra steps of setting up integration tests, regression tests artefacts configuration for questions, expected and responses. 

According to the Content team removal of remaining 2015/2016 will be done pretty soon (circa: February 2017).. Hence why I feel replacing the precalculate block with options for 2017/18 and 2016/17 will be just one change (ie. one file) and easier to follow.

**Comment by leenagupte** [comment]
@ikennaokpala This looks good 👍 

The only thing I would ask is to add a line to the commit note for ea0fb4c explaining that you have created new blank outcomes for '2017-2018' and that the content will be added in later commits. This will explain why there are empty test artefacts.

**Comment by ikennaokpala** [comment]
@leenagupte I have updated the commit 5007685dbec76a6a400607e2606c0fda18cf18df

**Pull request merged**